wandb/sdk/artifacts/artifact_saver.py

```diff
@@ -90,8 +90,12 @@ def save(
                 history_step,
                 base_id,
             )
+            return server_artifact, commit_fut
         finally:
-            self._cleanup_staged_entries()
+            if commit_fut is not None:
```

Comment:
`commit_fut` is not defined when `_save_internal` raises an exception:
![Untitled](https://github.com/wandb/wandb/assets/127154459/f0757205-7efc-4d5f-bdf2-d28ff5425a94)

Reply:
good catch, fixed